src/index.ts

```diff
@@ -97,6 +97,37 @@ app.event("reaction_added", async ({ body, client }) => {
   }
 });
 
+if (process.env.SLACK_AUTO_TRANSLATION_ENABLED === "1") {
+  app.message(async ({body, client}) => {
+
+    const event = body.event as MessageEvent;
+    if (event.subtype && event.subtype != 'message_replied' && event.subtype != 'me_message') {
+      console.debug(`Ignoring message with subtype ${event.subtype} from ${event.user}`)
+      return;
+    }
+    const channelId = event.channel;
+    const messageTs = event.ts;
+    if (!channelId || !messageTs) {
+      return;
+    }
+
+    const replies = await reacjilator.repliesInThread(client, channelId, messageTs);
+    if (replies.messages && replies.messages.length > 0) {
+      const message = replies.messages[0];
+      if (message.text) {
```

Comment:
You can skip the messages posted by this app. An easy way would be to check the poster's user ID. You can get the bot user's user ID. 

Also, the messages in a thread should come as a message event. I don't think you need to scan all the messages in a thread for each event.

Reply:
Yes I could remove that but as it is now, if somebody uses the flag but it's already auto-translated to that language it'll skip posting the translation.
Either way is fine here.